tests/integration_test/dynamic_routing_integration_test.go

```diff
@@ -720,3 +722,73 @@ func TestDynamicBackendRoutingMutualTLS(t *testing.T) {
 		}()
 	}
 }
+
+func TestDynamicGrpcBackendTLS(t *testing.T) {
+	t.Parallel()
+
+	configID := "test-config-id"
+	args := []string{"--service_config_id=" + configID, "--rollout_strategy=fixed"}
+
+	tests := []struct {
+		desc                string
+		clientProtocol      string
+		method              string
+		useWrongBackendCert bool
+		header              http.Header
+		wantResp            string
+		wantError           string
+	}{
+		{
+			desc:           "gRPC client calling gRPCs remote backend succeed",
+			clientProtocol: "grpc",
+			method:         "GetShelf",
+			header:         http.Header{"x-api-key": []string{"api-key"}},
+			wantResp:       `{"id":"100","theme":"Kids"}`,
+		},
+		{
+			desc:           "Http client calling gRPCs remote backend succeed",
+			clientProtocol: "http",
+			method:         "/v1/shelves/200?key=api-key",
+			wantResp:       `{"id":"200","theme":"Classic"}`,
+		},
+		{
+			desc:                "gRPC client calling gRPCs remote backend fail",
+			clientProtocol:      "grpc",
+			method:              "GetShelf",
+			useWrongBackendCert: true,
+			header:              http.Header{"x-api-key": []string{"api-key"}},
+			wantError:           "Unavailable",
```

Comment:
Nit: "503 Service Unavailable"

Reply:
This is gRPC, no HTTP status code returned.